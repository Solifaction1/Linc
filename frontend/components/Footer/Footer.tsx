import React from 'react'
import Image from 'next/image'
import { useMediaQuery } from 'react-responsive'
import Link from 'next/link'
import { Navigation } from '@models/navigation.model'
import { Localizations } from '@models/attributes.model'
import { BREAKPOINTS } from '@constants/breakpoints'
import DataCompliantLinks from './DataCompliantLinks/DataCompliantLinks'
import styles from './footer.module.scss'
import IconsRow from './IconsRow/IconsRow'
import InternalLinks from './InternalLinks/InternalLinks'

const Footer = ({
	internalLinksData,
	localizations,
}: {
	internalLinksData: Navigation[]
	localizations: Localizations | undefined
}) => {
	const isMobile = useMediaQuery({ maxWidth: BREAKPOINTS.md })

	const socialMediaItem = internalLinksData.find(
		(navigationitem) => navigationitem.path === 'social-media'
	)

	const contactItem = internalLinksData.find(
		(navigationitem) => navigationitem.path === 'contact'
	)

	const dataCompliantLinksItems = internalLinksData.filter(
		(navigationitem) => navigationitem.parent?.path === 'data-compliant-links'
	)

	return (
		<footer className={styles.footer}>
			<section className={styles.footerWrapper}>
				<div>
					<Link href="/">
						{isMobile ? (
							<Image
								src="/images/logo.svg"
								alt="logo"
								width={112}
								height={44}
								className={styles.logo}
								priority
								quality={60}
							/>
						) : (
							<Image
								src="/images/logo.svg"
								alt="logo"
								width={172}
								height={68}
								className={styles.logo}
								priority
								quality={60}
							/>
						)}
					</Link>
				</div>

				<InternalLinks
					internalLinksData={internalLinksData}
					localizations={localizations}
				/>

				<IconsRow
					contactItem={contactItem}
					facebookLink={socialMediaItem?.related?.facebook ?? ''}
					linkedInLink={socialMediaItem?.related?.linkedin ?? ''}
				/>
			</section>

			<section className={styles.dataCompliantLinksWrapper}>
				{dataCompliantLinksItems.length > 0 && (
					<DataCompliantLinks
						className={styles.dataCompliantLinks}
						dataCompliantLinks={dataCompliantLinksItems}
						localizations={localizations}
					/>
				)}
			</section>
		</footer>
	)
}

export default Footer
