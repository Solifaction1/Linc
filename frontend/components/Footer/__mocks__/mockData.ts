/* eslint-disable max-lines */
import { Contact } from "@models/contact.model"
import { Navigation, RelatedContactInformation } from "@models/navigation.model"

const mockNavigationParent1: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 1,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent1',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 1',
    type: '',
    uiRouterKey: 'parent1',
    updatedAt: ''
}

const mockNavigationParent2: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 4,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent2',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 2',
    type: '',
    uiRouterKey: 'parent2',
    updatedAt: ''
}


export const mockContact: Contact = {
    id: 10,
    street: 'street',
    houseNumber: 'houseNumber',
    postalCode: 0,
    city: 'city',
    telephone: '+1234567890'
}

const mockContactInformationRelated: RelatedContactInformation = {
    createdAt: 'createdAt',
    facebook: 'facebook',
    id: 7,
    instagram: 'instagram',
    linkedin: 'linkedin',
    locale: 'en',
    navigationItemId: 0,
    publishedAt: 'publishedAt',
    sitemap_exclude: false,
    updatedAt: 'updatedAt',
    updatedBy: 'updatedBy',
    __contentType: '__contentType'
}

const mockContactInformation: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 8,
    items: null,
    menuAttached: true,
    order: 0,
    parent: null,
    path: 'contact-information',
    related: mockContactInformationRelated,
    sitemap_exclude: true,
    title: 'Contact Information',
    type: '',
    uiRouterKey: 'contact-information',
    updatedAt: ''
}

const mockNavigationDataCompliantLinks: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 2,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'data-compliant-links',
    related: null,
    sitemap_exclude: true,
    title: 'Data Compliant Links',
    type: '',
    uiRouterKey: 'data-compliant-links',
    updatedAt: ''
}

export const mockNavigationData = [
    mockNavigationParent1,
    mockNavigationParent2,
    mockContactInformation,
    mockNavigationDataCompliantLinks,
]