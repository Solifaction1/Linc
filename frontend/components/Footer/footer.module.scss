@import "styles/vars";
@import 'styles/utilities';

.footer {
    background-color: $color-gray;

    .footerWrapper {
        @extend %page-container;
        padding-bottom: 0;
        
        @media (min-width: $break-xlg) {
            grid-template-columns: 40rem 4fr 6fr;
        }

        @media (max-width: $break-xlg) {
            grid-template-columns: 38rem 4fr 6fr;
        }

        @media (max-width: $break-md) {
            display: flex;
            flex-direction: column;
            padding-top: calc(#{$spacing-base} * 8);
        }

        .logo {
            margin-bottom: calc(#{$spacing-base} * 6);

            @media (max-width: $break-md) {
                margin-bottom: calc(#{$spacing-base} * 7.5);
            }
        }
        
        .internalLinksWrapper {
            @media (max-width: $break-md) {
                display: flex;
                flex-direction: column;
                margin-bottom: calc(#{$spacing-base} * 10);

                .languageSwitch {
                    button, a {
                        padding-top: 0;
                        padding-left: 0;

                        color: $color-white;

                        line-height: 2.1rem;
                        font-weight: $font-weight-regular;
                    }
                }
            }
        }

        .link {
            max-height: 3rem;
            margin-bottom: calc(#{$spacing-base} * 2);
            text-decoration: none;

            p {
                @extend %body-copy-3;
                color: $color-white;
                font-size: $font-size-text-md;
                line-height: 2.1rem;
            }
        }

        .externalLink {
            @media (max-width: $break-md) {
                margin-bottom: calc(#{$spacing-base} * 3);
            }
        }
    }

    .dataCompliantLinksWrapper {
        @extend %page-container;
        display: flex;
        padding-top: 0;
        padding-bottom: calc(#{$spacing-base} * 5);
    }
}
