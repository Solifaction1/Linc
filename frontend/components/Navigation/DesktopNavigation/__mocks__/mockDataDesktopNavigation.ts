/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    items: 'items',
    parent: null,
    related: null
}]
