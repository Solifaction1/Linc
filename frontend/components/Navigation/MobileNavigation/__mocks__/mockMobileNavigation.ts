/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    children: [],
    items: 'items',
    parent: null,
    related: {
        createdAt: '',
        facebook: 'https://www.facebook.com',
        linkedin: 'https://www.linkedin.com',
        instagram: '',
        locale: '',
        id: 1,
        navigationItemId: 2,
        publishedAt: '',
        sitemap_exclude: false,
        updatedAt: '',
        updatedBy: '',
        __contentType: ''
    }
}]
