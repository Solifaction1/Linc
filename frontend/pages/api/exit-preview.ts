export default function handler(
  _req: any,
  res: {
    clearPreviewData: () => void;
    status: (arg0: number) => {
      (): any;
      new (): any;
      end: { (): void; new (): any };
    };
  }
) {
  // Clear the cookies set by the preview mode.
  res.clearPreviewData();

  // Send a 200-Success response to the frontend
  res.status(200).end();
}
