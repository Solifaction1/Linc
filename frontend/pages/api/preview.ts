import { NextApiResponse, NextApiRequest } from 'next'
import { fetchContentPreviewApi } from '@services/api.service'

const handler = async (req: NextApiRequest, res: NextApiResponse) => {
	// Get the preview secret and the slug which needs to be previewed
	const { secret, slug } = req.query
	// If the secret passed as URL query parameter doesn't match the preview secret in .env
	// then send a 401-Unauthorized response
	if (secret !== process.env.STRAPI_PREVIEW_SECRET) {
		return res.status(401).json({ message: 'Invalid token' })
	}

	// If slug is not provided, send a 400-Bad Request response
	if (!slug) {
		return res.status(400).json({ message: 'Parameter `slug` is not provided' })
	}

	// Send a request to Strapi and fetch the article
	// to check if the provided slug exists
	const blog = await fetchContentPreviewApi(slug, 'blogs', 1)

	// If the article is not found, send a 404-Not Found response
	if (blog === null) {
		return res.status(404).json({ message: 'Blog not found' })
	}

	// Enable Preview Mode by setting the cookies
	res.setPreviewData({})

	// Redirect to the path of the article slug
	return res.redirect(307, `/blogs/${blog.data[0].attributes.slug}`)
}

export default handler
