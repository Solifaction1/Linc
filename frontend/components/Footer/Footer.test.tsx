import React from 'react'

import { render, waitFor } from '@testing-library/react'
import '@testing-library/jest-dom'
import { fetchCollectionApi } from '@services/api.service'
import Footer from './Footer'
import { mockNavigationData } from './__mocks__/mockData'

jest.mock('@services/api.service', () => ({
	fetchCollectionApi: jest.fn(),
}))
jest.mock('next/router', () => ({
	useRouter() {
		return {
			locale: 'locale',
			pathname: 'test/test',
		}
	},
}))

jest.mock('next-i18next', () => ({
	useTranslation() {
		return {
			t: () => {
				'test'
			},
		}
	},
}))

const mockedFetchCollectionApi = fetchCollectionApi as jest.Mock<unknown>

describe('Footer', () => {
	beforeEach(() => {
		mockedFetchCollectionApi.mockImplementation(() => mockNavigationData)
	})

	it('renders a Footer', async () => {
		const { container, getByText } = render(
			<Footer
				internalLinksData={mockNavigationData}
				localizations={{ data: undefined }}
			/>
		)

		expect(container.querySelector('footer')).toBeVisible()
		expect(container.getElementsByClassName('footerWrapper')[0]).toBeVisible()

		expect(container.querySelector('a[href="/"]')).toBeVisible()
		expect(container.getElementsByClassName('logo')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('iconsRow iconsRow')[0]
		).toBeVisible()

		await waitFor(() => {
			expect(container.getElementsByClassName('internalLinks')[0]).toBeVisible()
			expect(
				container.getElementsByClassName('link link link md false')[0]
			).toBeVisible()
			expect(container.querySelector('a[href="/parent1"]')).toBeVisible()
			expect(getByText('Parent 1', { selector: 'p' })).toBeVisible()

			expect(
				container.getElementsByClassName('link link link md false')[1]
			).toBeVisible()
			expect(container.querySelector('a[href="/parent2"]')).toBeVisible()
			expect(getByText('Parent 2', { selector: 'p' })).toBeVisible()

			expect(
				container.getElementsByClassName('link link link md false')[2]
			).toBeVisible()
			expect(
				container.querySelector('a[href="/contact-information"]')
			).toBeVisible()
			expect(getByText('Contact Information', { selector: 'p' })).toBeVisible()

			expect(
				container.getElementsByClassName('iconsRow undefined')[0]
			).toBeVisible()
			expect(container.getElementsByClassName('text')[0]).toBeVisible()

			expect(
				container.getElementsByClassName('dataCompliantLinksWrapper')[0]
			).toBeVisible()
		})

		expect(container).toMatchSnapshot()
	})
})
