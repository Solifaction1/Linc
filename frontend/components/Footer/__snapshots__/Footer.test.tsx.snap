// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders a Footer 1`] = `
<div>
  <footer
    class="footer"
  >
    <section
      class="footerWrapper"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="logo"
            class="logo"
            data-nimg="1"
            decoding="async"
            height="68"
            src="/images/logo.svg"
            srcset="/images/logo.svg 1x, /images/logo.svg 2x"
            style="color: transparent;"
            width="172"
          />
        </a>
      </div>
      <div
        class="internalLinks"
      >
        <a
          class="link link link md false"
          href="/parent1"
        >
          <p>
            Parent 1
          </p>
        </a>
        <a
          class="link link link md false"
          href="/parent2"
        >
          <p>
            Parent 2
          </p>
        </a>
        <a
          class="link link link md false"
          href="/contact-information"
        >
          <p>
            Contact Information
          </p>
        </a>
      </div>
      <div
        class="iconsRow undefined"
      >
        <p
          class="text"
        />
      </div>
    </section>
    <section
      class="dataCompliantLinksWrapper"
    />
  </footer>
</div>
`;
